app/views/layouts/_timeout.html

```diff
@@ -1,19 +1,22 @@
 <script>
   function PeriodicalQuery() {
-    var request = new XMLHttpRequest();
-    request.onload = function (event) {
-      if (request.response['timeout_status'] == 'no_current_user') {
-        return;
-      } else if (request.response['timeout_status'] == 'timeout_warning') {
-        $('#timeout-warning').slideDown('fast');
-      } else if (request.response['timeout_status'] == 'timed_out') {
-        window.location.href = '/users/start';
-      }
-    };
-    request.open('GET', '/timeout_check', true);
-    request.responseType = 'json';
-    request.send();
+    $.ajax({
```

Comment:
Sad performance, Internet Exploder.

Nice fix.  I agree with Navraj that I usually expect javascript in js files, but I think I remember from the last PR that you were including this html in an iframe?

Reply:
This is included in two places: the angular layout and the default Rails layout.  It's included in an HTML page, so we can and should keep it in the HTML format.